app/services/firebase/auth.ts

```diff
@@ -0,0 +1,46 @@
+import auth from '@react-native-firebase/auth';
+import _ from 'lodash';
+
+import { parseFairbaseError } from '../../utils';
+import { IFirebaseAuthError, IFirebaseAuthResponse } from './types';
+
+export const firebaseRegisterUser = async (email: string, password: string) => {
+  const registerUserResponse: IFirebaseAuthResponse = {
+    user: null,
+    error: null,
+  };
+
+  try {
+    const response = await auth().createUserWithEmailAndPassword(
+      email,
+      password,
+    );
+    registerUserResponse.user = _.cloneDeep(response.user);
+  } catch (error) {
+    console.log(typeof error);
+    registerUserResponse.error = parseFairbaseError(
+      (error as IFirebaseAuthError).code,
+    );
+  }
+  return registerUserResponse;
+};
+
+export const firebaseLoginUser = async (email: string, password: string) => {
+  const loginUserResponse: IFirebaseAuthResponse = {
+    user: null,
+    error: null,
+  };
+  try {
+    const response = await auth().signInWithEmailAndPassword(email, password);
+    loginUserResponse.user = _.cloneDeep(response.user);
```

Comment:
Let's do the next
```
loginUserResponse.user = await auth().signInWithEmailAndPassword(email, password).user;
```
The try/catch statement will save us if there will be an error